java/ql/src/semmle/code/java/frameworks/Networking.qll

```diff
@@ -19,6 +19,11 @@ class TypeUrl extends RefType {
   TypeUrl() { hasQualifiedName("java.net", "URL") }
 }
 
+/** The type `java.net.URI`. */
+class TypeUri extends RefType {
```

Comment:
There are 3 other `hasQualifiedName("java.net", "URI")` uses in the codebase -- let's factor them (in a separate commit) while we're introducing this

Reply:
Creating a separate PR to bring this change.

Reply:
See the new PR here. https://github.com/github/codeql/pull/4600